index.d.ts

```diff
@@ -1217,39 +1217,65 @@ declare module '@amaas/amaas-core-sdk-js' {
     namespace PositionPNL {
       function retrieve(
         { AMId }: { AMId: number },
```

Comment:
I think this is missing the query argument. Also, where it's known explicitly, let's set the type of the query rather than using `any`. e.g.

```typescript
function retrieve({ AMId, query }: {
  AMId: number,
  query: {
    bookId: string | string[],
    businessDate: string
  }
}) {
}
```

Also in general, can you ensure that the optionality is correctly defined. If it is defined as `query: {...}`, the TS compiler will not allow you to call `retrieve` without including the `query` argument. For this I believe it is fine, since `bookId` and `businessDate` is required, but further down in the insert I'm not sure if possible queryParams are mandatory.

Reply:
Also, `any | any[] == any`, since `any` could already be an array.